exercises/dolica_akelloegwel/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,173 @@
+#include <string>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <map>
+#include <cctype>
+#include <algorithm>
+#include <vector>
+#include <set>
+
+bool ContainsDashes(std::string compositeWord)
+{
+	return compositeWord.find("-") != std::string::npos;
+}
+std::vector<std::string> SplitWords(std::string compositeWord)
+{
+	// Containers for splitting up words
+	std::vector<std::string> words;
+	std::string segment;
+
+	// Convert the composite word to a stream
+	std::stringstream wordStream(compositeWord);
+
+	// Split the elements of the stream using the '-' character and place these in the vector
+	while (std::getline(wordStream, segment, '-'))
+	{
+		// Ignore empty words
+		if (segment.length() == 0)
+			continue;
+
+		// Add the word to the vector
+		words.push_back(segment);
+	}
+
+	return words;
+}
+std::string PrepareWord(std::string rawWord)
+{
+	// Declare a string of punctuation characters
+	std::string unwantedChars = ".,?'\"!():;";

```

Comment:
Could you do this using `boost::regex` i.e.`regex_search` I think this could be a useful further exercise. Would be a better solution that the character-by character match.